pallets/crowdloans/src/crowdloan_structs.rs

```diff
@@ -0,0 +1,281 @@
+                    value: amount,
+                    signature: None,
+                })),
+            ],
+        })));
+
+        let xcm = Xcm(vec![
+            WithdrawAsset(MultiAssets::from(asset.clone())),
+            BuyExecution {
+                fees: asset.clone(),
+                weight_limit: Unlimited,
+            },
+            Transact {
+                origin_type: OriginKind::SovereignAccount,
+                // TODO:
+                // determine correct weighting for this calll
+                require_weight_at_most: 100_000_000_000,
+                call: call.encode().into(),
+            },
+            RefundSurplus,
+            DepositAsset {
+                assets: asset.into(),
+                max_assets: 1,
+                beneficiary: T::AccountIdToMultiLocation::convert(
+                    T::SelfParaId::get().into_account(),
+                ),
+            },
+        ]);
+
+        // TODO:
+        // correctly remove the tokens from our Parachain.
+        // this possibly applys to fees and the transfer amount
+        // burning on one side and creating on the other is called
+        // teleporting in the Polkadot world
+        // T::Assets::burn_from(101, &T::PalletId::get().into_account(), amount)?;
+
+        // send xcm call
+        let _response = T::XcmSender::send_xcm(MultiLocation::parent(), xcm)
+            .map_err(|_| Error::<T>::SendXcmError)?;
+
+        Ok(())
+    }
+
+    fn withdraw<T: Config>(self, para_id: ParaId, _: CurrencyId) -> Result<(), DispatchError> {
+        let call_params = CrowdloanWithdrawCall::<T> {
+            who: T::SelfParaId::get().into_account(),
+            index: para_id,
+        };
+
+        let fees = 900_000_000_000;
+        frame_support::ensure!(!fees.is_zero(), Error::<T>::XcmFeesCompensationTooLow);
+        let asset: MultiAsset = (MultiLocation::here(), fees).into();
+
+        let contribute_call: DoubleEncoded<()> =
+            WestendCall::Crowdloan(CrowdloanCall::Withdraw(call_params))
+                .encode()
+                .into();
+
+        let xcm = Xcm(vec![
+            WithdrawAsset(MultiAssets::from(asset.clone())),
+            BuyExecution {
+                fees: asset.clone(),
+                weight_limit: Unlimited,
+            },
+            Transact {
+                origin_type: OriginKind::SovereignAccount,
+                require_weight_at_most: 100_000_000_000,
+                call: contribute_call,
+            },
+            RefundSurplus,
+            DepositAsset {
+                assets: asset.into(),
+                max_assets: 1,
+                beneficiary: T::AccountIdToMultiLocation::convert(
+                    T::SelfParaId::get().into_account(),
+                ),
+            },
+        ]);
+
+        // send xcm call
+        let _response = T::XcmSender::send_xcm(MultiLocation::parent(), xcm)
+            .map_err(|_| Error::<T>::SendXcmError)?;
+
+        Ok(())
```

Comment:
This could be replaced with a simple `T::XcmSender::send_xcm(MultiLocation::parent(), xcm).map_err(|_| Error::<T>::SendXcmError)`